pkg/volume/host_path/host_path.go

```diff
@@ -123,8 +123,12 @@ func (plugin *hostPathPlugin) NewUnmounter(volName string, podUID types.UID) (vo
 	}}, nil
 }
 
-func (plugin *hostPathPlugin) NewRecycler(pvName string, spec *volume.Spec) (volume.Recycler, error) {
-	return plugin.newRecyclerFunc(pvName, spec, plugin.host, plugin.config)
+func (plugin *hostPathPlugin) Recycle(pvName string, spec *volume.Spec) error {
+	recycler, err := plugin.newRecyclerFunc(pvName, spec, plugin.host, plugin.config)
```

Comment:
Is there a reason to keep this as a func that returns a func which is immediately called?  Or can we move the contents of the returned recyclerFunc into here and remove a few levels of indirection?